frame/support/src/storage/mod.rs

```diff
@@ -716,6 +774,42 @@ pub struct PrefixIterator<T> {
 }
 
 impl<T> PrefixIterator<T> {
+	/// Creates a new `PrefixIterator`, starting from `next_key` and filtering out keys that are
+	/// not prefixed with `prefix`.
+	///
+	/// A `decode_fn` function must also be supplied, and it takes in two `&[u8]` parameters,
+	/// returning a `Result` containing the decoded type `T` if successful, and a `codec::Error` on
+	/// failure. The first `&[u8]` argument represents the raw, undecoded key without the prefix of
+	/// the current item, while the second `&[u8]` argument denotes the corresponding raw,
+	/// undecoded value.
+	pub fn new(
+		prefix: Vec<u8>,
+		next_key: Vec<u8>,
+		decode_fn: fn(&[u8], &[u8]) -> Result<T, codec::Error>,
+	) -> Self {
+		PrefixIterator {
+			prefix,
+			previous_key: next_key,
+			drain: false,
+			closure: decode_fn,
+		}
+	}
+
+	/// Get the last key that has been iterated upon and return it.
+	pub fn next_key(&self) -> &[u8] {
+		&self.previous_key
+	}
+
+	/// Get the prefix that is being iterated upon for this iterator and return it.
+	pub fn prefix(&self) -> &[u8] {
+		&self.prefix
+	}
+
+	/// Set the next key that the iterator should start iterating upon.
+	pub fn set_next_key(&mut self, next_key: Vec<u8>) {
```

Comment:
same I feel like `next_key` is a confusing name, `set_last_raw_key` or `set_current_raw_key` is a better name IMO